pkg/gui/types/common.go

```diff
@@ -45,6 +45,7 @@ type IGuiCommon interface {
 	CurrentContext() Context
 	CurrentStaticContext() Context
 	IsCurrentContext(Context) bool
+	ActivateContext(context Context) error
```

Comment:
I wasn't totally sure if this is the best way to fix the issue, but it felt ok to me. I do realize that this is another one that conflicts with your big refactor...